kratos/includes/model_part.h

```diff
@@ -584,6 +614,157 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
+
+    MasterSlaveConstraint::Pointer CreateNewMasterSlaveConstraint(const std::string& ConstraintName, 
+                                                                                    IndexType Id, 
+                                                                                    NodeType& rMasterNode,
+                                                                                    const DoubleVariableType& rMasterVariable,
+                                                                                    NodeType& rSlaveNode,
+                                                                                    const DoubleVariableType& rSlaveVariable,
+                                                                                    const double Weight,
+                                                                                    const double Constant);
+
+    MasterSlaveConstraint::Pointer CreateNewMasterSlaveConstraint(const std::string& ConstraintName, 
+                                                                                    IndexType Id, 
+                                                                                    NodeType& rMasterNode,
+                                                                                    const VariableComponentType& rMasterVariable,
+                                                                                    NodeType& rSlaveNode,
+                                                                                    const VariableComponentType& rSlaveVariable,
+                                                                                    double Weight,
+                                                                                    double Constant);
+
+    /** Remove the master-slave constraint with given Id from mesh with ThisIndex in this modelpart and all its subs.
+    */
+    void RemoveMasterSlaveConstraint(IndexType MasterSlaveConstraintId);
+
+    /** Remove given master-slave constraint from mesh with ThisIndex in this modelpart and all its subs.
+    */
+    void RemoveMasterSlaveConstraint(MasterSlaveConstraintType& ThisMasterSlaveConstraint);
+
+    /** Remove the master-slave constraint with given Id from mesh with ThisIndex in parents, itself and children.
+    */
+    void RemoveMasterSlaveConstraintFromAllLevels(IndexType MasterSlaveConstraintId);
+
+    /** Remove given master-slave constraint from mesh with ThisIndex in parents, itself and children.
+    */
+    void RemoveMasterSlaveConstraintFromAllLevels(MasterSlaveConstraintType& ThisMasterSlaveConstraint);
+
+    /** Returns the MasterSlaveConstraint::Pointer  corresponding to it's identifier */
+    MasterSlaveConstraintType::Pointer pGetMasterSlaveConstraint(IndexType ConstraintId);
+
+    /** Returns a reference MasterSlaveConstraint corresponding to it's identifier */
+    MasterSlaveConstraintType& GetMasterSlaveConstraint(IndexType MasterSlaveConstraintId);
```

Comment:
get methods should have a const overload

Reply:
done